### ISSUE TITLE
Haddock link to Char type points to GHC.Types module instead of Data.Char

### ISSUE THREAD

**Issue by yaitskov** [haskell/haskell-language-server#3470]
```
2023-01-28T00:04:07.439064Z | Info | No log file specified; using stderr.
haskell-language-server version: 1.9.0.0 (GHC: 9.2.4) (PATH: /nix/store/ayy98lcs1v8ip4mryjxi11hb8diqibh0-haskell-language-server-1.9.0.0/bin/.haskell-language-server-9.2.4-unwrapped)

```

**Comment by Bodigrim** [comment]
Context: https://github.com/haskell/core-libraries-committee/issues/123#issuecomment-1407203872

**Comment by georgefst** [comment]
> Context: [haskell/core-libraries-committee#123 (comment)](https://github.com/haskell/core-libraries-committee/issues/123#issuecomment-1407203872)

Yes, as mentioned in that comment, this is a much wider problem than this thread title suggests. Rather than the "special treatment" fix suggested there, _ideally_ HLS would do something more general. After all, we know which module any identifier was actually imported from, so in theory it shouldn't be too hard to refer there instead of the original definition. Another rule-of-thumb might be that we shouldn't be pointing to packages which aren't explicitly in scope e.g. `ghc-prim` for most users of `base`.
